File: Java version/app/src/main/java/it/unical/mat/embasp/specializations/solver_planning_domains/SPDPlan.java
package it.unical.mat.embasp.specializations.solver_planning_domains;

import it.unical.mat.embasp.languages.pddl.Plan;
import it.unical.mat.parsers.pddl.PDDLSolversParser;

public class SPDPlan extends Plan {

	public SPDPlan(final String plan, final String error) {
		super(plan, error);
	}

	@Override
	protected void parse() {
		if((errors != null && !errors.isEmpty()) || (output != null && output.isEmpty()))
		    return;
		
		errors += PDDLSolversParser.parseSolverPlanningDomains(this, output, true);
	}

}

File: CSharp version/it/unical/mat/embasp/specializations/solver_planning_domains/SPDPlan.cs
using it.unical.mat.parsers.pddl;

namespace it.unical.mat.embasp.specializations.solver_planning_domains
{
    using Plan = it.unical.mat.embasp.languages.pddl.Plan;

    public class SPDPlan : Plan
    {
        public SPDPlan(string plan, string error) : base(plan, error) { }

        protected internal override void Parse()
        {
            if (!string.IsNullOrEmpty(errors) || string.IsNullOrEmpty(output))
                return;

            errors += PDDLSolversParser.ParseSolverPlanningDomains(this, output, true);
        }
    }
}